File: system/multitier/backend-java/src/main/java/com/mycompany/myshop/backend/core/dtos/BrowseOrderHistoryResponse.java
package com.mycompany.myshop.backend.core.dtos;

import com.mycompany.myshop.backend.core.entities.OrderStatus;
import lombok.Data;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

@Data
public class BrowseOrderHistoryResponse {
    private List<BrowseOrderHistoryItemResponse> orders;

    @Data
    public static class BrowseOrderHistoryItemResponse {
        private String orderNumber;
        private Instant orderTimestamp;
        private String sku;
        private String country;
        private int quantity;
        private BigDecimal totalPrice;
        private OrderStatus status;
        private String appliedCouponCode;
    }
}

File: system/multitier/backend-dotnet/Core/Dtos/BrowseOrderHistoryResponse.cs
using MyCompany.MyShop.Backend.Core.Entities;

namespace MyCompany.MyShop.Backend.Core.Dtos;

public class BrowseOrderHistoryResponse
{
    public List<BrowseOrderHistoryItemResponse> Orders { get; set; } = new();
}

public class BrowseOrderHistoryItemResponse
{
    public string OrderNumber { get; set; } = null!;
    public DateTime OrderTimestamp { get; set; }
    public string Sku { get; set; } = null!;
    public string Country { get; set; } = null!;
    public int Quantity { get; set; }
    public decimal TotalPrice { get; set; }
    public OrderStatus Status { get; set; }
    public string? AppliedCouponCode { get; set; }
}
